ansible/roles/monasca_swarm_service/templates/monasca_swarm_service.yml.j2

```diff
@@ -0,0 +1,83 @@
+---
+version: "3.6"
+
+networks:
+  hostnet:
+    external: true
+    name: host
+
+configs:
+  fluentd:
+      file: {{ tmp_folder.path }}/fluentd.conf
+
+services:
+  fluentd:
+    image: stackhpc/monasca-fluentd
+    networks:
+      hostnet: {}
+    deploy:
+      mode: global
+    configs:
+      - source: fluentd
+        target: "/fluentd/etc/fluent.conf"
+        mode: 0644
+
+  monasca-forwarder:
+    image: stackhpc/agent-forwarder
+    hostname: "{% raw %}{{.Node.Hostname}}{% endraw %}"
+    networks:
+      hostnet: {}
+    deploy:
+      mode: global
+    environment:
+      - "LOG_LEVEL={{ monasca_swarm_service_log_level }}"
+      - "OS_AUTH_URL={{ monasca_swarm_service_keystone_uri }}"
+      - "OS_USERNAME={{ monasca_swarm_service_username }}"
+      - "OS_PASSWORD={{ monasca_swarm_service_password }}"
+      - OS_USER_DOMAIN_NAME=Default
+      - "OS_PROJECT_NAME={{ monasca_swarm_service_project_name }}"
+      - OS_PROJECT_DOMAIN_NAME=Default
+      - "MONASCA_URL={{ monasca_swarm_service_api_uri }}"
+      - SERVICE_TYPE=monitoring
+      - ENDPOINT_TYPE=public
+      - REGION_NAME=RegionOne
+      - "FORWARDER_URL=http://127.0.0.1:{{ monasca_swarm_service_forwarder_port }}"
+      - "FORWARDER_PORT={{ monasca_swarm_service_forwarder_port }}"
+
+  monasca-collector:
+    image: stackhpc/agent-collector
+    hostname: "{% raw %}{{.Node.Hostname}}{% endraw %}"
+    networks:
+      hostnet: {}
+    deploy:
+      mode: global
+    environment:
+      - HOST=true
+      - DOCKER=true
+      - DOCKER_ROOT=/
+      - DOCKER_SOCKET=unix://var/run/docker.sock
+      - "LOG_LEVEL={{ monasca_swarm_service_log_level }}"
+      - "FORWARDER_URL=http://127.0.0.1:{{ monasca_swarm_service_forwarder_port }}"
+      - "FORWARDER_PORT={{ monasca_swarm_service_forwarder_port }}"
+    volumes:
+      - "/:/rootfs"
+      - "/var/run:/var/run:rw"
+      - "/sys:/sys:ro"
+      - "/var/lib/docker/:/var/lib/docker:ro"
+      - "/dev/disk/:/dev/disk:ro"
```

Comment:
I've occasionally seen weird issues with stopping and removing containers when mounting in special places like this. Just one to check.

Reply:
Updated so that DOCKER_ROOT points to /rootfs mounted in from the host. Good spot @markgoddard.